### ISSUE TITLE
Rocketchat Jira oauth integration

### ISSUE THREAD

**Issue by rgoerner** [RocketChat/Rocket.Chat#14557]
Hi,

I am playing around to reach my goal to identify all users and roles from the internal Jira-Directory as Hipchat did before. We do not want to use the Crowd from Atlassian (I know there is support on it in Rocketchat already).

I did the following:

- in Jira, I Created a incoming-only application link:
-> url = base url of our rocketchat-instance
-> consumer_key = access_token
-> public_key as explained here: https://developer.atlassian.com/server/jira/platform/jira-rest-api-example-oauth-authentication-6291692/
-> consumer-callback-URL as provided in rocketchat oauth custom area

- in rocketchat, the following settings were done:
-> URL: base-URL of Jira
-> Token-Path base-URL of Jira/plugins/servlet/oauth/access-token
-> Token send over User-Data (also tried with Header)
-> Authorization-Path: base-URL of Jira/plugins/servlet/oauth/authorize
-> param name of token: access_token
-> private key: the private key of the public key which is provided at the Jira-instance 
-> field for username: username

However, I get the following error:

WARN anonymous 743x1356589x1 191gj9b 185.38.48.138 /plugins/servlet/oauth/authorize [c.a.o.s.i.servlet.authorize.AuthorizeServlet] Problem encountered authenticating OAuth client for url "https:// base-URL of Jira /plugins/servlet/oauth/authorize", error was "parameter_absent", with parameters "{oauth_parameters_absent=oauth_token, oauth_problem=parameter_absent}"

Can anyone help me?

We can try all together and then, would have a new Oauth-Plugin maybe for Rocketchat which would help others, too

Thanks
Ronny

**Comment by geekgonecrazy** [comment]
It looks like jira is expecting some non-standard paramaters to their oauth endpoints.

it wants the parameter oauth_token and Rocket.Chat isn't sending that. 

If you take a look at the url your browser goes to when you click the button you'll see the params sent.

Looking at the params it wants.. you'll likely have to implement an oauth provider inside Rocket.Chat to make it work.  I don't think you'll be able to use our custom oauth feature :(

**Comment by rgoerner** [comment]
@geekgonecrazy yes, this is correct. But rocketchat does not support custom RSA Auth and the parameter "param name of the token" seems to be not working correctly, too, or I am too silly ;-)

**Comment by rgoerner** [comment]
I tried to avoid this problem, but now I was running into another issue, that I can not run my own iframe-integration on the app for Android (known issue). So :( @geekgonecrazy  can you tell me the file, where the normal login process does appear that I can have a look into the code? Thanks

**Comment by geekgonecrazy** [comment]
Take a look at any of the current authentication methods added:

https://github.com/RocketChat/Rocket.Chat/tree/develop/app/dolphin
https://github.com/RocketChat/Rocket.Chat/tree/develop/app/drupal
https://github.com/RocketChat/Rocket.Chat/tree/develop/app/gitlab

There are a few
